--- a/conda/gateways/disk/lock.py
+++ b/conda/gateways/disk/lock.py
@@ -7,6 +7,7 @@ between conda processes. Try to acquire a lock on a single byte in the metadat
 file; modify both files; then release the lock.
 """
 
+import sys
 import time
 import warnings
 from contextlib import contextmanager
@@ -86,7 +87,7 @@ except ImportError:
 
 
 def lock(fd, *, lock_attempts=LOCK_ATTEMPTS):
-    if not context.no_lock:
+    if not context.no_lock and sys.platform != "emscripten":
         # locking required for jlap, now default for all
         return _lock_impl(fd, lock_attempts)
     return _lock_noop(fd, lock_attempts)
